### PR TITLE
added a note to explain hour placeholder in icp = next (T-??)

### LINKED ISSUE #70: cylc user guide: more advanced cycling info
We could do with more information about advanced cycling features such as:
- min(...)
- truncation syntax (e.g. `T-00` for `start at the next 00 minutes`)

and anything else that comes to mind.

### LINKED ISSUE #367: initial cycle point: explain T-?? syntax
In the documentation for the previous/next syntax (initial cycle point) we should add a note or explaining sentence to cover the `T-??` syntax.

See https://github.com/cylc/cylc-doc/issues/19#issuecomment-499054436

### PR DESCRIPTION
**Pull request by wxtim**
Closes #367 & #455
May close #70 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.

### CODE REVIEW

File: src/user-guide/writing-workflows/scheduling.rst [hunk 1]
```diff
@@ -537,6 +537,10 @@ A list of times, separated by semicolons, can be provided, e.g.
 brackets, and if more than one is given, the major time unit in each (hours
 or minutes) should be of the same type.
 
+.. note::
+
+   In this context ``T-00`` "-" is a placeholder for the hours column.
```
Comment (oliver-sanders) [inline src/user-guide/writing-workflows/scheduling.rst:542]: Slightly confusing, the `T-` is a placeholder for hours in any context. Perhaps something like:

```suggestion
   ``T-00`` means every hour, on the hour. The ``-`` is a placeholder for the hours column.
```

File: src/user-guide/writing-workflows/scheduling.rst [hunk 2]
```diff
@@ -574,21 +578,20 @@ Note that day-of-week should always be specified when using weeks. If a time
...
-   ====================================  ==================
+   Syntax, Description, Interpretation
+   ``next(-00)``,                     Any century; next year 00,                  2100-01-01
+   ``previous(--01)``,                Any year; next month 01,                    2018-01-01
+   ``next(---01)``,                   Any year; any month; next 1st of month,     2018-04-01
+   ``previous(--1225)``,              Any year; previous Dec 25,                  2017-12-25
+   ``next(-2006)``,                   Any century; next June in a year ending 20, 2020-06-01
+   ``previous(-W101)``,               Any century; previous week 10 day 1,        2018-03-05
+   ``next(-W-1; -W-3; -W-5)``,        "Any year; any week; next day 1, 3 or 5",   2018-03-14
+   ``next(-001; -091; -181; -271)``,  "Any year; day 1, 91, 181 or 271",          2018-04-01
```
Comment (oliver-sanders) [inline src/user-guide/writing-workflows/scheduling.rst:593]: FYI the `list-table` directive is more flexible (supporting any arbitrary ReST) and doesn't suffer from the need to quote commas.

https://docutils.sourceforge.io/docs/ref/rst/directives.html#list-table

**Review by oliver-sanders — APPROVED** [review]

**Comment by MetRonnie** [comment]
Can you rebase onto 8.0.x please

**Review by datamel — APPROVED** [review]
Built and checked. Good to go from me, just one small query.

**Pull request merged**